Python/instrumentation.c

```diff
@@ -1605,10 +1581,14 @@ _Py_Instrument(PyCodeObject *code, PyInterpreterState *interp)
         return 0;
     }
     /* Insert instrumentation */
+    int code_len = (int)Py_SIZE(code);
     for (int i = code->_co_firsttraceable; i < code_len; i+= _PyInstruction_GetLength(code, i)) {
         _Py_CODEUNIT *instr = &_PyCode_CODE(code)[i];
         CHECK(instr->op.code != 0);
         int base_opcode = _Py_GetBaseOpcode(code, i);
+        if (base_opcode == ENTER_EXECUTOR) {
```

Comment:
Please let me know if we want to maintain this code as an assert statement.

Reply:
As written this is definitely wrong, this function should only return `-1` if an exception is set.

I think it is appropriate an an `assert()` -- on L1554 above you have cleared executors, which involves restoring all the original opcodes that were shadowed by `ENTER_EXECUTOR`, so at this point in the code we expect no executors to exist. If any do exist, something's wrong (maybe no-GIL :-) and it would be nice to have a failing `assert()` instead of a random crash.

Reply:
> As written this is definitely wrong, this function should only return -1 if an exception is set.

Ah, yes I missed it.

> I think it is appropriate an an assert()

+1